File: code/dsa/linked_lists/python/zip_recursive.py
def zip_linked_list(head1, head2):
    if head1 is None and head2 is None:
        return None
    if head1 is None and head2 is not None:
        return head2
    if head1 is not None and head2 is None:
        return head1
    next_head1 = head1.next
    next_head2 = head2.next
    head1.next = head2
    head2.next = zip_linked_list(next_head1, next_head2)
    return head1

File: code/dsa/linked_lists/javascript/zip_recursive.js
function zipLinkedList(head1, head2) {
    if ((head1 === null) && (head2 === null)) return null;
    if ((head1 === null) && (head2 !== null)) return head2;
    if ((head1 !== null) && (head2 === null)) return head1;

    let head1_next = head1.next;
    let head2_next = head2.next;
    head1.next = head2.next;
    head2.next = zipLinkedList(head1_next, head2_next);
    return head1;
}
